common/stub-setup.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Injects the stubs into the global scope.
+ * Bundled HomeApp cannot be loaded until these globals are set.
+ */
+import { DeviceManagerStub } from './device-manager';
+import { AppStub } from './smart-home-app';
+
+(global as any).smarthome = {
+  App: AppStub,
+  DeviceManager: DeviceManagerStub,
+};
+
+//TODO(cjdaly) find a cleaner way to do this.
+export function loadHomeApp(path: string) {
+  // Implicitly runs the JavaScript module at `path`
+  require(path);
```

Comment:
You might want to leverage the vm module https://nodejs.org/api/vm.html to have a little bit more controlled isolation between the platform and the app.

Alternatively something like https://www.npmjs.com/package/web-worker could help you formalize the message passing interface between the platform and the app (similar to the sample implementation I shared before https://gist.github.com/proppy/e728e1eff87020295d2b77014fef4e2a#file-server-ts-L88).

Reply:
I opened an issue for this (#22)